src/document.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Sajari;
+
+class Key
+{
+    /** @var $field string */
+    private $field;
+    private $value;
+
+    /**
+     * Key constructor.
+     * @param $field string
+     * @param $value
+     */
+    public function __construct($field, $value)
+    {
+        $this->field = $field;
+        $this->value = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getField()
+    {
+        return $this->field;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    /**
+     * @return \sajari\engine\store\doc\Keys\Key
```

Comment:
The proto should never be exposed. The problem is I don't see a nice way to have the proto stuff available on all the objects (to make it easy to construct requests) but not expose it to the end user.